redis/conn.go

```diff
@@ -127,6 +142,15 @@ func DialNetDial(dial func(network, addr string) (net.Conn, error)) DialOption {
 	}}
 }
 
+// DialNetDialContext specifies a custom dial function with context for creating TCP
+// connections, otherwise a net.Dialer customized via the other options is used.
+// DialNetDialContext overrides DialConnectTimeout and DialKeepAlive.
+func DialNetDialContext(dialContext func(ctx context.Context, network, addr string) (net.Conn, error)) DialOption {
+	return DialOption{func(do *dialOptions) {
+		do.dialContext = dialContext
```

Comment:
We could simplify things by only using a dialer with context, as mentioned above.

Reply:
Fix!